huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -312,7 +312,7 @@ func getProductDetail(config *config.Config, d *schema.ResourceData, engineType
 		}
 		for _, v := range ps.Values {
 			for _, p := range v.Details {
-				if engineType == "kafka" {
+				if v.Name == "single" || engineType == "kafka" {
```

Comment:
This seems a bit difficult for understanding, how about just checking rabbitmq engineType and not single for response with ProductInfos like below?
```
if engineType == "rabbitmq" && v.Name == "cluster"
```

Reply:
I added a note for this code

Reply:
Although this works, I think we should write it like this
```
if engineType == "kafka" {
    ...
} else if engineType == "rabbitmq" {
    if v.Name == "single" {
        ...
    } else {
        ...
    }
}
```
to make the logic clear

Reply:
The function has been split into two functions